scipy/signal/spectral.py

```diff
@@ -141,7 +141,7 @@ def periodogram(x, fs=1.0, window=None, nfft=None, detrend='constant',
                  scaling, axis)
 
 
-def welch(x, fs=1.0, window='hann', nperseg=256, noverlap=None, nfft=None,
+def welch(x, fs=1.0, window='hann', nperseg=None, noverlap=None, nfft=None,
```

Comment:
The changes in the default values need to be reflected in the docstrings. For example, `noverlap` also defaults to `None` and the docstring states: "If None, `noverlap = nperseg // 2`.  Defaults to None."